esp/templates/program/modules/studentregtwophase/rank_classes.html

```diff
@@ -73,5 +73,9 @@
     <i class="star-fill icon-2x icon-star" data-bind="visible: interested"></i>
   </a>
   <label>Click the star to mark this class as interested.</label>
+  <br />
+  {% for pri, pri_display in priorities %}
+  <button data-bind="click: function(data) { $parent.prioritySelection[{{pri}}-1](data.id); $j('#pri-{{pri}}').trigger('change.select2'); }">Set as {{pri_display}}</button>
```

Comment:
I could definitely put it in `esp/public/media/scripts/catalog-new.js`. I didn't really want to, though, because that script is used in a couple templates and this function is really tightly coupled to this particular templates, having to trigger the custom `change.select2` event on our special `#pri-{{pri}}` element (which doesn't exist in other templates using that JS file) for it to sync back up with knockout.js updating the computed checkbox.